app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -7,12 +7,43 @@ def make_offer
         vendor_api_token_id: @current_vendor_api_token.id,
       )
 
-      decision = MakeAnOffer.new(
-        actor: api_user,
-        application_choice: application_choice,
-        offer_conditions: params.dig(:data, :conditions),
-        course_data: params.dig(:data, :course),
-      )
+      course_data = params.dig(:data, :course)
+
+      if course_data.present?
+        course_option = GetCourseOptionFromCodes.new(
+          provider_code: course_data[:provider_code],
+          course_code: course_data[:course_code],
+          recruitment_cycle_year: course_data[:recruitment_cycle_year],
+          study_mode: course_data[:study_mode],
+          site_code: course_data[:site_code],
+        ).call
+
+        change_offer_form = ProviderInterface::ChangeOfferForm.new step: :update,
```

Comment:
It feels a bit suspect that this caller needs to know about the steps in the `ChangeOfferForm` in a different namespace.  Could we somehow pull the part which is actually shared into an independent service?

Reply:
Yes, I think you're right. I compared the logic of `GetCourseOptionForCodes` and `ChangeOfferForm` and they both check that all attributes have been set and correspond to each other. The only additional value we get from using `ChangeOfferForm` here is a check on whether the new course is `open_on_apply`. So, I'll add an explicit check for that in the controller and remove this `ChangeOfferForm` dependency.